src/ess/reduce/nexus/workflow.py

```diff
@@ -242,8 +245,34 @@ def get_transformation_chain(
 
 def to_transformation(
     chain: NeXusTransformationChain[Component, RunType],
+    interval: TimeInterval[RunType],
+    time_filter: NeXusTransformationTimeFilter[Component],
 ) -> NeXusTransformation[Component, RunType]:
     """Convert transformation chain into a single transformation matrix."""
+
+    chain = deepcopy(chain)
+    for t in chain.transformations.values():
+        # TODO Check if value is DataArray?
+        if t.sizes == {}:
+            continue
+        start = interval.value.start
+        stop = interval.value.stop
+        if isinstance(start, sc.Variable) or isinstance(stop, sc.Variable):
+            # NXlog entries are generally interpreted as the previous value being valid
+            # until the next entry. We therefore need to select the previous value, and
+            # any index after the last entry refers to the last entry, i.e., there is no
+            # "end" time in the files. We add a dummy end so we can use Scipp's label-
+            # based indexing for histogram data.
+            time = t.value.coords['time']
+            delta = sc.scalar(86_400_000, unit='s', dtype='int64').to(unit=time.unit)
```

Comment:
I think that is tricky, since we don't know the input dtype (could be signed or unsigned, or a datetime). There probably is a way (can you think of a simple one?), but just adding 1000 days seemed "safe".

Reply:
I think it is safe enough. I was more surprised by the concrete number and wondered whether it has some significance because it is not simply `10**10` or something like that.